molgenis-data/src/main/java/org/molgenis/data/support/MapEntity.java

```diff
@@ -61,7 +61,8 @@ public void set(Entity other, EntityMetaData metaData)
 		this.idAttributeName = entityMetaData.getIdAttribute().getName();
 		for (AttributeMetaData attribute : metaData.getAtomicAttributes())
 		{
-			set(attribute.getName(), other.get(attribute.getName()));
+			if (Iterables.contains(other.getAttributeNames(), attribute.getName())) set(attribute.getName(),
```

Comment:
This change is what breaks the annotator unit tests. `other` is a mock and getAttributeNames returns null.